addon/plugins/citation-plugin/index.ts

```diff
@@ -269,14 +275,20 @@ function collectDecorations(
   decsToAdd: Decoration[],
   schema: CitationSchema,
   regex: RegExp = CITATION_REGEX,
+  doc: PNode,
   decsToRemove?: Decoration[],
   oldDecs?: DecorationSet
 ) {
   return function (node: PNode, pos: number): boolean {
+    const resolvedNode = doc.resolve(pos);
     if (
       node.isText &&
       node.text &&
-      !schema.marks.citation.isInSet(node.marks)
+      !(
+        resolvedNode &&
+        resolvedNode.parent &&
+        resolvedNode.parent.type === schema.nodes.link
+      )
```

Comment:
In here you could check if the resolvedNode has any ancestor of type link.

Reply:
Like this better? :)

Reply:
You can also use `findParentNodeOfTypeClosestToPos`, you can apply that function on a resolved position, then you don't need to create a node-selection.